### ISSUE TITLE
Implement HTTP Caching

### ISSUE THREAD

**Issue by pl12133** [hak8or/WeatherSpot#11]
[Google Fundamentals: HTTP Caching](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching?hl=en)

With successful caching scheme our pages should usually be returning **304: Not Modified** instead of **200: OK**.

**Comment by pl12133** [comment]
I must be missing something here, I've set the Cache-Control header for our apache server but it seems things still aren't being cached. Changes are live in apache2.conf on our server, adding Cache-Control header for a list of file types.

[HTTP Advanced Request Header Tool](http://www.askapache.com/online-tools/http-headers-tool/)
[Cache Control Header Checker](http://highloadtools.com/cachecontrol)

If you type our site into either one of these tools you can see the line `Cache-Control: max-age=172800, public, must-revalidate` in the headers; but **200 OK** is still being returned instead of **304 Not Modified** and caching isn't happening.

**Comment by pl12133** [comment]
I was missing the Expires module, I loaded the expires module into mods_available on our server and added an "Expires" header alongside the Cache-Control headers; caching should now work.

I checked in Developers Tools that the page is loading from cache when I request it, but requests still return a **200 OK** instead of **304 Not Modified** which bothers me because returning **304** would save network latency. However the page should now be a little faster for the user.
